src/lib.rs

```diff
@@ -28,23 +42,62 @@ use log::Level;
 /// ```
 #[derive(StructOpt, Debug, Clone)]
 pub struct Verbosity {
-    /// Pass many times for more log output
-    ///
-    /// By default, it'll only report errors. Passing `-v` one time also prints
-    /// warnings, `-vv` enables info logging, `-vvv` debug, and `-vvvv` trace.
-    #[structopt(long = "verbosity", short = "v", parse(from_occurrences))]
+    /// Pass many times for less log output. see `--log-level`
+    #[cfg(not(feature = "quiet"))]
+    #[structopt(
+        short = "q", long = "quiet", group = "clap_verbosity_flag", parse(from_occurrences)
+    )]
+    quietness: u8,
+
+    /// Pass many times for more log output. see `--log-level`
+    #[cfg(not(feature = "trace"))]
+    #[structopt(
+        short = "v", long = "verbose", group = "clap_verbosity_flag", parse(from_occurrences)
+    )]
     verbosity: u8,
+
+    /// Set log level. By default it is info [possible values: quiet, error,
+    /// warn, info, debug, trace]
+    ///
+    /// Alternatively It's possible to use `-v`, `-vv` to increase and `-q`,
+    /// `-qq` etc. to decrease the log level.
+    #[structopt(
+        long = "log-level",
+        group = "clap_verbosity_flag",
+        raw(possible_values = "VERBOSITY_LEVELS", hide_possible_values = "true")
+    )]
+    level: Option<String>,
 }
 
 impl Verbosity {
     /// Get the log level.
-    pub fn log_level(&self) -> Level {
-        match self.verbosity {
-            0 => Level::Error,
-            1 => Level::Warn,
-            2 => Level::Info,
-            3 => Level::Debug,
-            _ => Level::Trace,
+    pub fn log_level(&self) -> Option<Level> {
+        if let Some(level) = self.level.as_ref() {
+            match level.as_ref() {
+                "quiet" => None,
+                "error" => Some(Level::Error),
+                "warn" => Some(Level::Warn),
+                "info" => Some(Level::Info),
+                "debug" => Some(Level::Debug),
+                "trace" => Some(Level::Trace),
+                _ => unreachable!(),
+            }
+        } else {
+            #[cfg(feature = "quiet")]
+            let verbosity = DEFAULT_VERBOSITY + self.verbosity;
+            #[cfg(feature = "trace")]
+            let verbosity = DEFAULT_VERBOSITY.saturating_sub(self.quietness);
+            #[cfg(not(any(feature = "quiet", feature = "trace")))]
+            let verbosity = (DEFAULT_VERBOSITY + self.verbosity).saturating_sub(self.quietness);
```

Comment:
I have changed this but I think this is not what you meant. The problem is that the Struct field disapears with the flag otherwise i could just add them even if one is always zero and be done with it.